thali/thalicryptomanager.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+var crypto = require('crypto');
+var fs = require('fs');
+var path = require('path');
+
+// The password is not secure because anyone who can get to the file can get
+// to the app and thus can get the password. The password is used here only to
+// satisfy the crypto/PKCS12 APIs.
+var password = 'password';
+var certname = 'certname';
+var country = 'country';
+var organization = 'organization';
+var pkcs12FileName = '/pkcs12.pfx';
+var macName = 'SHA256';
+var hashSizeInBytes = 16;
+
+/**
+* Checks if a PKCS12 file exists in a known location and if not present, it is
+* created. The public key is extracted from the PKCS12 content and it's SHA256
+* hash value is returned.
+* @param {Function} cb the callback which returns an error or the hash value.
+*/
+exports.getPublicKeyHash = function (cb) {
+  // get the path where the PKCS12 file is saved
+  Mobile.GetDocumentsPath(function (err, fileLocation) {
+    if (err) {
+      console.error("GetDocumentsPath err: ", err);
+      cb(err);
+    } else {
+      var file = path.join(fileLocation, pkcs12FileName);
+      getPKCS12Content(file, function(err, pkcs12Content) {
+        if(err) {
+          console.error('failed to get pkcs12 content');
+          cb(err);
+        }
+        
+        try {
+          var publicKey = crypto.pkcs12
+            .extractPublicKey(password, pkcs12Content);
+          // check if the extracted public key is good
+          if ( !publicKey || publicKey.length <= 0) {
+            console.error('extracted public key is invalid');
+            cb('extracted public key is invalid');
+          }
+          var hash = generateSlicedSHA256Hash(publicKey, hashSizeInBytes);
+          cb(null, hash);
+        } catch(e) {
+          console.error('error thrown by extractPublicKey() function');
+          cb('error thrown by extractPublicKey() function');
+        }
+      });
+    }
+  });
+};
+
+exports.getSlicedSHA256Hash = function(stringValueToHash, hashSizeInBytes) {
```

Comment:
Why would you change the function name? Why not just export generatedSlicedSHA256Hash directly? What was the motivation?